commands/utils/git.go

```diff
@@ -334,6 +354,24 @@ func (gm *GitManager) dryRunClone(destination string) error {
 	return nil
 }
 
+// Construct HTTPS clone url from the provided git info.
+// Frogbot already has an access token with sufficient permissions to clone with HTTPS,
+// in case we encounter SSH clone url, we generate HTTPS url instead.
+func (gm *GitManager) generateHTTPSCloneUrl() (url string, err error) {
+	switch gm.git.GitProvider {
+	case vcsutils.GitHub:
+		return fmt.Sprintf(githubHttpsFormat, gm.git.RepoOwner, gm.git.RepoName), nil
+	case vcsutils.GitLab:
+		return fmt.Sprintf(gitLabHttpsFormat, gm.git.GitProject, gm.git.RepoName), nil
+	case vcsutils.BitbucketServer:
+		return fmt.Sprintf(bitbucketServerHttpsFormat, gm.git.ApiEndpoint, gm.git.RepoOwner, gm.git.RepoName), nil
+	case vcsutils.AzureRepos:
+		return fmt.Sprintf(azureDevopsHttpsFormat, gm.git.RepoOwner, gm.git.RepoOwner, gm.git.GitProject, gm.git.RepoName), nil
+	default:
+		return "", fmt.Errorf("unsupported version control provider: %s", gm.git.GitProvider.String())
+	}
+}
+
```

Comment:
This would not work for on-premises. We need to support such cases based on the given `ApiEndpoint`. We can fallback to the cloud format if the `ApiEndpoint` is empty.